dali/kernels/tensor_shape.h

```diff
@@ -381,7 +381,9 @@ struct TensorListShapeBase {
   TensorShape<tensor_ndim> tensor_shape(int64_t sample) const {
     static_assert(tensor_ndim == sample_ndim || sample_ndim == DynamicDimensions
                   || tensor_ndim == DynamicDimensions, "Cannot convert to other static ndim");
-    assert(tensor_ndim == sample_dim() && "Cannot convert to other ndim");
+    if (tensor_ndim != DynamicDimensions && sample_ndim != DynamicDimensions) {
```

Comment:
```suggestion
    if (tensor_ndim != DynamicDimensions) {
```
Even if sample_ndim is dynamic you still want to check if static value of tensor_ndim matches current sample_dim().